data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/ArtifactRequestMessageSender.java

```diff
@@ -0,0 +1,149 @@
+package org.eclipse.dataspaceconnector.ids.api.multipart.dispatcher;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.fraunhofer.iais.eis.ArtifactRequestMessage;
+import de.fraunhofer.iais.eis.ArtifactRequestMessageBuilder;
+import de.fraunhofer.iais.eis.DynamicAttributeTokenBuilder;
+import de.fraunhofer.iais.eis.Message;
+import de.fraunhofer.iais.eis.TokenFormat;
+import jakarta.ws.rs.core.MediaType;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Headers;
+import okhttp3.HttpUrl;
+import okhttp3.MultipartBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.eclipse.dataspaceconnector.ids.spi.IdsIdParser;
+import org.eclipse.dataspaceconnector.ids.spi.IdsType;
+import org.eclipse.dataspaceconnector.ids.spi.spec.extension.ArtifactRequestMessagePayload;
+import org.eclipse.dataspaceconnector.ids.transform.IdsProtocol;
+import org.eclipse.dataspaceconnector.spi.iam.IdentityService;
+import org.eclipse.dataspaceconnector.spi.message.MessageContext;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public class ArtifactRequestMessageSender implements IdsMessageSender<DataRequest> {
+    private final ObjectMapper objectMapper;
+    private final Monitor monitor;
+    private final OkHttpClient okHttpClient;
+    private final IdentityService identityService;
+    // TODO Create Container ID class to ensure its correctly build
+    private final String connectorId;
+    private final String connectorEndpoint;
+
+    public ArtifactRequestMessageSender(@NotNull Monitor monitor, @NotNull String connectorId, @NotNull String connectorEndpoint, @NotNull ObjectMapper objectMapper, @NotNull OkHttpClient okHttpClient, @NotNull IdentityService identityService) {
```

Comment:
Does it make sense to have a message sender that can only ever send to one endpoint, i.e. the `connectorEndpoint`?

Reply:
That's one of the draft implementations, I used to test my code. But it will be replaces by the ArtifactRequestMessage sender from #275